attendance/lib/main.dart

```diff
@@ -1,31 +1,46 @@
+import 'package:attendance/AttendanceDatailsPage/attendance_details_page.dart';
+import 'package:attendance/ProfilePage/profile_page.dart';
+import 'package:attendance/homePage/home_page.dart';
+import 'package:attendance/loginPage/login_page.dart';
+import 'package:attendance/offline_page/offline_page.dart';
 import 'package:flutter/material.dart';
 
-import './AttendanceDatailsPage/attendance_details_page.dart';
-import './ProfilePage/profile_page.dart';
-import './homePage/home_page.dart';
-import './loginPage/login_page.dart';
-import './offline_page/offline_page.dart';
 
-void main() => runApp(new MyApp());
+void main() => runApp(MyApp());
 
+/// The Application class
 class MyApp extends StatelessWidget {
-  final Widget myHome = LoginPage();
+  final Widget _myHome = LoginPage();
 
   @override
-  Widget build(BuildContext context) {
-    return new MaterialApp(
+  Widget build(BuildContext context) =>
+    MaterialApp(
       title: 'Attendance',
-      theme: new ThemeData(
+      theme: ThemeData(
         primarySwatch: Colors.blue,
       ),
-      home: myHome,
+      home: _myHome,
+      onGenerateRoute: _getRoute,
       routes: <String, WidgetBuilder>{
-        '/login': (BuildContext context) => new LoginPage(),
-        '/home': (BuildContext context) => new HomePage(),
-        '/details': (BuildContext context) => new AttendanceDetailsPage(),
-        '/offline': (BuildContext context) => new OfflinePage(),
-        '/profile': (BuildContext context) => new ProfilePage()
+        '/login': (BuildContext context) => LoginPage(),
+        '/home': (BuildContext context) => HomePage(),
+        '/offline': (BuildContext context) => OfflinePage(),
+        '/profile': (BuildContext context) => const ProfilePage()
```

Comment:
@ammaratef45 Is that 'const' here on purpose?

Reply:
@Bahaaib The linter found that it's suggested.
Still investigating why it thought so...